source/CF/Helpers.pm

```diff
@@ -275,7 +275,7 @@ sub is_bam_paired_end {
 	my $header_found;
 	foreach (split("\n", $headers)){
 		if(/^\@PG/){
-			if(/-1/ && /-2/){
+			if(/\s+-1\s+/ && /\s+-2\s+/){

```

Comment:
This entire function has been replaced in my current version:

```perl
sub is_bam_paired_end {

	# Load samtools
	my @modules = ('samtools');
	my %loaded_mods = ();
	&CF::Helpers::load_environment_modules(\@modules,\%loaded_mods);

	my ($file) = @_;

	unless($file =~ /.bam$/ || $file =~ /.sam$/){
		warn "\n$file is not a .bam or .sam file - can't figure out PE / SE mode..\nExiting..\n\n";
		die;
	}

    # Read the first 1000 lines withx samtools
    my $se_reads = 0;
    my $pe_reads = 0;
    my $readcount = 0;
    open(my $fh, '-|', "samtools view $file") or die "Could not run samtools to check BAM pe/se: $!";
    while (<$fh>) {
        last if ($readcount >= 1000);
        my ($flag) = (split (/\t/))[1];
        if ($flag & 0x1){
            $pe_reads++;
        } else {
            $se_reads++;
        }
        $readcount++;
    }
    close($fh);

    # Look at our counts
    if($pe_reads >= 800){
        return 1;
    } else {
        return 0;
    }

}
```

Any objections to me ignoring your change and using my version of the function?